algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -144,145 +114,119 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
     auto init_categories = kernel_factory.getKernel("init_categories", &st);
     DAAL_CHECK_STATUS_VAR(st);
 
-    const size_t nDataBlocks = nDataRows / dataBlockSize + int (nDataRows % dataBlockSize != 0);
-    const size_t nProbeBlocks = nProbeRows / probeBlockSize + int (nProbeRows % probeBlockSize != 0);
-    QuickSelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), categories.type(), &st);
+    const size_t nDataBlocks  = nDataRows / dataBlockSize + int(nDataRows % dataBlockSize != 0);
+    const size_t nProbeBlocks = nProbeRows / probeBlockSize + int(nProbeRows % probeBlockSize != 0);
+    SelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), &st);
     DAAL_CHECK_STATUS_VAR(st);
 
-    BlockDescriptor<algorithmFpType> probeRows;
-    DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(0, nProbeRows, readOnly, probeRows));
-    auto probes_all = probeRows.getBuffer();
-
-    BlockDescriptor<algorithmFpType> dataRows;
-    DAAL_CHECK_STATUS_VAR(points->getBlockOfRows(0, nDataRows, readOnly, dataRows));
-    auto data_all = dataRows.getBuffer();
-
-    BlockDescriptor<int> labelRows;
-    DAAL_CHECK_STATUS_VAR(labels->getBlockOfRows(0, nDataRows, readOnly, labelRows));
-    auto labels_all = labelRows.getBuffer();
+    SelectIndexed::Params params(nK, TypeIds::id<algorithmFpType>(), dataBlockSize, parameter->engine);
+    SelectIndexedFactory factory;
+    SharedPtr<SelectIndexed> selector(factory.Create(nK, params, &st));
```

Comment:
Suppose you can not create a `selector` when `Create` method fails. 
Suggest to decompose it:
```cpp
auto something = factory.Create(nK, params, &st);
DAAL_CHECK_STATUS_VAR(st);
SharedPtr<SelectIndexed> selector (something);
```
Please also rename `Create()` to `create()`

Reply:
Is `factory.Create()` always returns `nullptr` when an error occurs?